schema/connector-discovery-schema.json

```diff
@@ -0,0 +1,297 @@
+            "$ref": "#/definitions/input_field",
+            "description": "Each additional property describes an input field."
+          }
+        },
+        "pollable": {
+          "type": "boolean",
+          "description": "A hint for the client that polling for this object_type might produce objects containing new data."
+        },
+        "metadata": {
+          "type": "object",
+          "description": "Free form key-value pairs. Can contain additional metadata about the object_type, if anything is required by client."
+        },
+        "endpoint": {
+          "$ref": "#/definitions/link",
+          "description": "Link to an endpoint that produces this object_type."
+        }
+      },
+      "required": [
+        "endpoint"
+      ]
+    },
+    "input_field": {
+      "type": "object",
+      "properties": {
+        "regex": {
+          "type": "string",
+          "description": "Regular expression that can be used by client to capture valid inputs, from a text line or paragraph."
+        },
+        "capture_group": {
+          "type": "integer",
+          "description": "Which parentheses group of the regex captured text to be used."
+        },
+        "env": {
+          "type": "string",
+          "description": "To be collected from client's Environment variables and send in the request. See Discovery in wiki."
+        },
+        "required": {
+          "type": "boolean",
+          "description": "A flag says if this input field is mandatory. If client knows this field then it can include in the request for better results."
+        }
+      },
+      "dependencies": {
+        "regex": {
+          "required": [
+            "capture_group"
+          ]
+        }
+      }
+    },
+    "actions": {
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/connector_level_action",
+        "description": "Each additional property is a connector level action."
+      }
+    },
+    "connector_level_action": {
+      "type": "object",
+      "properties": {
+        "url": {
+          "$ref": "#/definitions/link",
+          "description": "Link to the action."
+        },
+        "user_input": {
+          "type": "array",
+          "description": "Allows the user to enter data to be submitted in this action.",
+          "items": {
+            "$ref": "#/definitions/connector_action_user_input"
+          }
+        },
+        "fields": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/input_field",
+            "description": "Each additional property describes an input field."
+          }
+        },
+        "label": {
+          "type": "object",
+          "description": "Localised human-readable description of the action."
+        },
+        "completed_label": {
+          "type": "object",
+          "description": "A Localised message that says if the action was success in the backend."
```

Comment:
To be precise, both `label` and `completed_label` are text of the action buttons

Reply:
ok, Thanks. Then I think I should help users understand that - this will be the text of the button after action is completed successfully.
- "A localized message that says action was successful. Upon success the button text will show this label."

Reply:
It would also be good to mention that the locale is the key (ex. en-US, en-GB, en-IN, etc.) and the message is the value.

Reply:
`label` & `completed_label` are of type `string` not `object` (for card level action though which boxer supports)

Reply:
These are not card actions though.  These are connector-level-actions that are served by discovery.  Discovery does not take in a Lang header, so the json will not be localized.  This is similar to the config defined in discovery, where you have to specify the locale as a key, and the string as the value, so multiple locales can be sent to the client.